### DISCUSSION TITLE
Why not use useCallback with empty dependency instead of create new useMemoizedFn hook?

### DISCUSSION THREAD

**Discussion by nanghadiios** [alibaba/hooks#2344] [General]
After researching about `useMemoizedFn` hook in ahooks, I have a question that needs to be explained, why not use `useCallback` with empty dependency for persistent functions?

ex:

```
const [state, setState] = useState('');

// When the state changes, the func reference will never change
const func = useCallback(() => {
  console.log(state);
}, []);
```

**Comment by liuyib** [comment]
See:
- Issues: https://github.com/alibaba/hooks/issues/697
- PR: https://github.com/alibaba/hooks/pull/712

After the above changes, the usage of `useMemoizedFn` is similar to [`useEffectEvent`](https://react.dev/reference/react/experimental_useEffectEvent), but also have a little difference. In one way, it can memoize a function, and in the other way, it can avoid the problem of closures (similar to `useEffectEvent`), so there is no need to pass the second parameter (dependencies array).

I hope this answer can help you.
